server/src/Settings.ts

```diff
@@ -631,6 +632,73 @@ export class Settings {
         return timeout;
     }
 
+    /**
+     * Searches for a Java home and tries to use it.
+     * Promis is resolved with the path to the Java executable that should be used.
+     * Otherwise, promise is rejected with an error message (as string) in case something went wrong
+     */
+    public static async getJavaPath(): Promise<string> {
+        const configuredJavaBinary = Settings.settings.javaSettings.javaBinary;
+        const searchForJavaHome = configuredJavaBinary == null || configuredJavaBinary == "";
+        let javaPath: string
+        if (searchForJavaHome) {
+            // no java binary configured, search for it:
+            const javaHome = await Settings.getJavaHome();
+            Log.log(`Java was successfully located at ${javaHome.executables.java}`, LogLevel.Debug);
+            javaPath = javaHome.executables.java;
+        } else {
+            Log.log(`Uses Java home found in settings: ${configuredJavaBinary}`, LogLevel.Debug);
+            javaPath = configuredJavaBinary;
+        }
+
+        // try to execute `java -version`:
+        try {
+            const javaVersionOutput = await Common.spawn(javaPath, ["-version"]);
+            const javaVersion = javaVersionOutput.stdout.concat(javaVersionOutput.stderr);
+            Log.log(`Java home found: ${javaPath}. It's version is: ${javaVersion}`, LogLevel.Verbose);
+            return javaPath;
+        } catch (err) {
+            let errorMsg: string
+            if (searchForJavaHome) {
+                errorMsg = `A Java home was found at '${javaPath}' but executing it with '-version' has failed: ${err}.`;
+            } else {
+                errorMsg = `The Java home is in the settings configured to be '${javaPath}' but executing it with '-version' has failed: ${err}.`;
+            }
+            // rethrow error
+            throw errorMsg;
+        }
+    }
+
+    /**
+     * Searches for a Java home. Promise is rejected with an error message (as string) in case something went wrong
+     */
+    private static getJavaHome(): Promise<IJavaHomeInfo> {
+        return new Promise((resolve, reject) => {
+          try {
+            const options = {
+              version: ">=1.8",
+              mustBe64Bit: true
+            };
+            locate_java_home.default(options, (err, javaHomes) => {
+              if (err) {
+                reject(err.message);
+              } else {
+                if (!Array.isArray(javaHomes) || javaHomes.length === 0) {
+                  const msg = "Could not find a 64-bit Java installation with at least version 1.8. "
+                    + "Please install one and/or manually specify it in the Gobra settings.";
+                  reject(msg);
+                } else {
+                  const javaHome = javaHomes[0];
```

Comment:
Minor: Maybe it is worth warning the user that we arbitrarily picked one of the available Java installations.

Reply:
Would you log a warning? I'm a bit hesitant to just insert a `Log.hint` here as I do not want to spam the user with popup. Also, this function seems to get called when checking the settings and when starting ViperServer, thus a potential popup might appear multiple times (that also why error messages do not directly get logged but returned as a promise failure)

Reply:
The scenario I have in mind is one that actually happened on one of our customer's laptop some time ago. Assume the user has multiple Java installations, e.g. one Java 8 64-bit without the server component and one Java 8 64-bit *with* the server component. If the IDE automatically picks the former, ViperServer startup command will crash with a not-100%-obvious stack trace. The user might then try running the java commend in their command line, but that might actually succeed if the right java version is on $PATH. This would seem to the user as a bug in the IDE, whereas (in my view) it's actually their fault since they didn't specify which Java version the IDE should use. Therefore, in such a scenario I would think it's good practice to be as explicit as possible, even showing a popup that explains how to change the settings in such a way that would solve the problem and the warning would not appear anymore. 

How about something like this: 
```
Warning: multiple Java installations detected: 
  /usr/lib/java   [automatically selected]
  /User/Library/Oracle/Java 11/Home
To suppress this warning, please define 
"viper.javaSettings.javaBinary" in your settings file. 
```

At the very least, I suggest logging this information via ```Log.log(msg, LogLevel.Info)```. However, I don't fully understand why the popup would appear multiple times, precisely because this should only be checked at IDE startup time.

Reply:
I've now implemented it by extending the settings check and display a corresponding warning in such cases. As it turns out, my own setup is affected by the ambiguity and thus I've checked that the warning is correctly displayed :P 
@aterga You might want to have a quick look at the `checkSettings` rewrite, which is now `async` and hopefully a bit easier to read

Reply:
Great, thanks! I will merge this PR very soon. 

Is there an easy way to link the Settings button in the Java-installation-warning popup to lead to the **JSON** version of the settings, rather than the GUI? GUI settings aren't currently supported by Viper IDE, so ideally we should direct the user to the JSON file (if VS Code API provides this feature).

Reply:
Not sure whether this would be possible. One thing that might complicate things is that to the best of my knowledge there can be different places to store the settings file (user global vs. workspace settings). Thus, I don't think we should attempt to reimplement this logic

Reply:
I am not suggesting to reimplement the logic. But currently the button already leads to some GUI wrapper over a JSON file. Ideally, it should lead to the file itself, not the GUI, because users who are unfamiliar with VS Code might not know how to get to the JSON in which they can control Viper's settings.